tests/app/service/test_rest.py

```diff
@@ -991,6 +1026,45 @@ def test_update_service_permissions_will_add_service_permissions(client, sample_
     assert set(result['data']['permissions']) == set([SMS_TYPE, EMAIL_TYPE, LETTER_TYPE])
 
 
+@pytest.mark.parametrize(
+    'email_from, should_error',
+    (
+            ('sams.sarnies', False),  # We are happy with plain ascii alnum/full-stops.
+            ('SAMS.SARNIES', True),  # We will reject anything with uppercase characters
+            ('sam\'s.sarnies', True),  # We reject punctuation other than a full-stop `.`
+            ('sams.Бутерброды', True),  # We reject unicode outside of the ascii charset
+            ('sams.ü', True),  # Even if it could theoretically be downcast to ascii
+            ('sams.u', False),  # Like this, which would be fine
+    )
+)
+def test_update_service_allows_only_lowercase_digits_and_fullstops_in_email_from(
+        client, sample_service, email_from, should_error
+):
+    auth_header = create_admin_authorization_header()
+
+    data = {
+        'service_name': 'Sam\'s sarnies',
+        'email_from': email_from
+    }
+
+    resp = client.post(
```

Comment:
This test could also use the `admin_request` fixture

Reply:
It does now :)